src/main/java/htsjdk/samtools/util/AbstractLocusIterator.java

```diff
@@ -0,0 +1,572 @@
+     * as well as useIndex==false, and generally will be much faster.
+     */
+    private final boolean useIndex;
+
+    /**
+     * LocusInfos on this list are ready to be returned by iterator.  All reads that overlap
+     * the locus have been accumulated before the AbstractLocusInfo is moved into this list.
+     */
+    private final ArrayList<K> complete = new ArrayList<>(100);
+
+    /**
+     * LocusInfos for which accumulation is in progress.  When {@link #accumulateSamRecord(SAMRecord)} is called
+     * the state of this list is guaranteed to be either:
+     * a) Empty, or
+     * b) That the element at index 0 corresponds to the same genomic locus as the first aligned base
+     * in the read being accumulated
+     * <p>
+     * Before each new read is accumulated the accumulator is examined and:
+     * i) any LocusInfos at positions earlier than the read start are moved to {@link #complete}
+     * ii) any uncovered positions between the last AbstractLocusInfo and the first aligned base of the new read
+     * have LocusInfos created and added to {@link #complete} if we are emitting uncovered loci
+     */
+    final ArrayList<K> accumulator = new ArrayList<>(100);
+
+    private int qualityScoreCutoff = Integer.MIN_VALUE;
+    private int mappingQualityScoreCutoff = Integer.MIN_VALUE;
+    private boolean includeNonPfReads = true;
+
+    /**
+     * If true, emit a AbstractLocusInfo for every locus in the target map, or if no target map,
+     * emit a AbstractLocusInfo for every locus in the reference sequence.
+     * If false, emit a AbstractLocusInfo only if a locus has coverage.
+     */
+    private boolean emitUncoveredLoci = true;
+
+    /**
+     * If set, this will cap the number of reads we accumulate for any given position.
+     * Note that if we hit the maximum threshold at the first position in the accumulation queue,
+     * then we throw further reads overlapping that position completely away (including for subsequent positions).
+     * This is a useful feature if one wants to minimize the memory footprint in files with a few massively large pileups,
+     * but it must be pointed out that it could cause major bias because of the non-random nature with which the cap is
+     * applied (the first maxReadsToAccumulatePerLocus reads are kept and all subsequent ones are dropped).
+     */
+    private int maxReadsToAccumulatePerLocus = Integer.MAX_VALUE;
+
+    /**
+     * Set to true when we have enforced the accumulation limit for the first time
+     */
+    private boolean enforcedAccumulationLimit = false;
+
+    /**
+     * If true, include indels in the LocusInfo
+     */
+    protected boolean includeIndels = false;
+
+    /**
+     * When there is a target mask, these members remember the last locus for which a AbstractLocusInfo has been
+     * returned, so that any uncovered locus in the target mask can be covered by a 0-coverage AbstractLocusInfo
```

Comment:
Right.  We think it is not reasonable to change it to K, because K extends AbstractLocusInfo.